### PR TITLE
Demo - Stupid repository implementation

### PR DESCRIPTION
**Pull request by basvodde**
This is an example of how you can implement the repository. Not done yet.

### CODE REVIEW

**Comment by arstrube** [comment]
Okay, looking good so far. Could you please use the established order for the arguments in `copy()` though?

**Comment by arstrube** [comment]
Now I'm looking closer, the order of arguments is actually correct. But I find the naming really confusing. To me, `out` is and **output** parameter and `in` is an **input** parameter. I would prefer something like `src` and `dst` or even just `obj1` and `obj2` which at least doesn't lead me to wrong assumptions.

**Comment by arstrube** [comment]
Looking good so far. 

I deleted a bunch of comments I made earlier, because somehow I got it all wrong. Not quite awake yet? I wonder....

File: include/CppUTestExt/MockNamedValue.h [hunk 1]
```diff
@@ -42,6 +42,16 @@ class MockNamedValueComparator
     virtual SimpleString valueToString(const void* object)=0;
 };
 
+class MockNamedValueCopier
+{
+public:
+    MockNamedValueCopier() {}
+    virtual ~MockNamedValueCopier() {}
+
+    virtual void copy(const void* in, void* out)=0;
```
Comment (arstrube) [inline include/CppUTestExt/MockNamedValue.h]: Please put the output parameter first. Doing so is a common idiom used in many places, SimpleString among others.

**Comment by arstrube** [comment]
I did a head-to-head comparison on my fork, by pushing a similar "stupid" implementation of the 2-list approach. I concede that the one-list implementation is preferable:
+ It has fewer changes, overall
+ It is simpler
+ It doesn't need ClearComparators() and ClearCopiers(), which I never really liked
+ It can keep clear(), as opposed to having clearAll()

I would like to suggest the following small improvements (for which I can provide PRs)
+ Follow the standard idiom regarding the arguments of copy()
+ Use of _override in the test cases 
+ Add a test case to demonstrate that, when a comparator and a copier are installed for the same type, both will be found
+ Use "MyType" as typename rather than "copier", which is misleading

**Comment by basvodde** [comment]
@arstrube I'm fine with all suggestions. I just wrote the stupid version to show that implementing it was really easy :)

So, considering that the implementation *is* really easy? Do you and @jgonzalezdr want to decide between the three solutions. The choices are:

1) two separate repositories
2) One repository with two lists
3) One repository with one list

What do you prefer?

**Comment by arstrube** [comment]
Cunning :) 

I actually prefer 3) now I've seen it, obiously with the suggested improvements ;-)

Is there still a chance to vote for "Operator" instead of ComparatorsAndCopiers, perhaps? 

The stupid version isn't that stupid, really -- there isn't a whole lot missing to complete the MockNamedValue aspect of things. (can do PR for that).

**Comment by basvodde** [comment]
Heh, it was stupid because it doesn't store the Comparator and the Copier in the same node. That would make it a bit more complicated... not a lot though. Just the inserting would change.

**Comment by arstrube** [comment]
Oh, what a shame. I am really disappointed that you want to make it more complicated. That way, you will lose your advantage over the two-lists competition. Maybe I'll revise my vote, after all.....

It works perfectly fine as is, although perhaps, inelegantly. I added a test case that proves it. They don't have to be in the same node. I thought that was very clever of you once I realized that. Now you call it stupid =(

**Comment by arstrube** [comment]
Having to search the nodes for inserting was one of the major drawbacks I saw in your proposal. You don't want to retrofit something like that now, do you?

**Comment by arstrube** [comment]
My vote now goes to:

4) One repository with one _stupid_ list

Bah =)

**Comment by arstrube** [comment]
So I grafted the remainder of our changes and tests onto the _simple_ (not stupid) list and all works just fine.

The only thing I dislike is that there are still two lists in MockSupport_c; would need to put in a bit more thought on how to get rid of that.

**Comment by arstrube** [comment]
How about the MockSupport interface. Currently, we have: `RemoveAllComparators()`.
Do we just add `RemoveAllCopiers()` (this would require corresponding repository methods)?
Or do we do the otherwise preferable `RemoveAllComparatorsAndCopiers()`and break the interface?

**Comment by basvodde** [comment]
I suggest RemoveAllCompiers.

**Comment by arstrube** [comment]
o.k.

**Pull request merged**